warp/src/raygun/mod.rs

```diff
@@ -334,127 +334,262 @@ impl Ord for MessagePage {
+            Conversation::Group(group) => &group.common,
         }
     }
 }
 
 impl Conversation {
-    pub fn id(&self) -> Uuid {
-        self.id
+    pub fn set_id(mut self, id: Uuid) -> Self {
+        self.common_mut().id = id;
+
+        self
     }
 
-    pub fn name(&self) -> Option<String> {
-        self.name.clone()
+    pub fn id_mut(&mut self) -> &mut Uuid {
+        &mut self.common_mut().id
     }
 
-    pub fn creator(&self) -> Option<DID> {
-        self.creator.clone()
+    pub fn set_name(mut self, name: Option<String>) -> Self {
+        self.common_mut().name = name;
+
+        self
     }
 
-    pub fn created(&self) -> DateTime<Utc> {
-        self.created
+    pub fn name_mut(&mut self) -> &mut Option<String> {
+        &mut self.common_mut().name
     }
 
-    pub fn modified(&self) -> DateTime<Utc> {
-        self.modified
+    pub fn set_creator(mut self, creator: Option<DID>) -> Self {
+        self.common_mut().creator = creator;
+
+        self
     }
 
-    pub fn conversation_type(&self) -> ConversationType {
-        self.conversation_type
+    pub fn creator_mut(&mut self) -> &mut Option<DID> {
+        &mut self.common_mut().creator
     }
 
-    pub fn settings(&self) -> ConversationSettings {
-        self.settings
+    pub fn set_created(mut self, created: DateTime<Utc>) -> Self {
+        self.common_mut().created = created;
+
+        self
     }
 
-    pub fn recipients(&self) -> Vec<DID> {
-        self.recipients.clone()
+    pub fn created_mut(&mut self) -> &mut DateTime<Utc> {
+        &mut self.common_mut().created
+    }
+
+    pub fn set_modified(mut self, modified: DateTime<Utc>) -> Self {
+        self.common_mut().modified = modified;
+
+        self
+    }
+
+    pub fn modified_mut(&mut self) -> &mut DateTime<Utc> {
+        &mut self.common_mut().modified
+    }
+
+    pub fn set_recipients(mut self, recipients: Vec<DID>) -> Self {
+        self.common_mut().recipients = recipients;
+
+        self
+    }
+
+    pub fn add_recipient(mut self, recipient: DID) -> Self {
+        self.common_mut().recipients.push(recipient);
+
+        self
+    }
+
+    pub fn recipients_mut(&mut self) -> &mut Vec<DID> {
+        &mut self.common_mut().recipients
+    }
+
+    fn common_mut(&mut self) -> &mut ConversationCommon {
+        match self {
+            Conversation::Direct(direct) => &mut direct.common,
+            Conversation::Group(group) => &mut group.common,
+        }
     }
 }
 
-impl Conversation {
-    pub fn set_id(&mut self, id: Uuid) {
-        self.id = id;
+impl Default for Conversation {
+    fn default() -> Self {
+        Self::Direct(DirectConversation::default())
     }
+}
 
-    pub fn set_name(&mut self, name: Option<String>) {
-        self.name = name;
+#[derive(Debug, Clone, Serialize, Deserialize, Eq)]
+pub struct DirectConversation {
+    #[serde(flatten)]
+    common: ConversationCommon,
+    settings: DirectConversationSettings,
```

Comment:
As mentioned, I do feel this should be removed since direct conversations would not really have any permissions or settings of sort.

Reply:
> As mentioned, I do feel this should be removed since direct conversations would not really have any permissions or settings of sort.

As long as we're certain of this but I'd like this to go in a separate PR. I'm doing a lot in here, as it is. :)

Reply:
It can be done in a separate PR